docs/AA--Javascript-in-EE/value-objects/label.md

```diff
@@ -0,0 +1,168 @@
+## Label Value Object
+The label value object represents a label which consists of a singular string and plural string.  For example a shoe label where singular is `shoe` and plural is `shoes`.
+
+## Features
+
+The label value object exposes the singular and plural forms of the labels as their raw value but also allows for retrieving formatted strings via helper methods.
+
+## Installation/Loading
+
+The label vo is available via the `eejs` global on its `valueObjects` property.  So you can access it directly via `eejs.valueObjects.Label` provided the bundled `@eventespresso/vo` javascript has been loaded before your script. For instructions on different ways to do this you can [go here](README.md#Usage).
+
+## API
+
+### Constructor
+A `Label` value object can be constructed with valid singular and plural strings passed in as the arguments.  If a non string type is passed in for either singular or plural arguments, then a `TypeError` exception is thrown.
+
+**Example:**
+
+```js
+const myShoeLabel = new Label( 'shoe', 'shoes' );
+```
+
+### _Methods_
+#### `Label.assertString`
+This is used to assert the passed in value is a string.  It throws a `TypeError` when the value is not a string.
+
+#### `Label.fromSameSingleAndPlural`
+This is used to construct an instance of Label using the passed in argument for both singular and plural forms of the label.  This is useful when a label has no distinction between its singular and plural usage.
+
+**Example:**
+
+```js
+const approvedLabel = Label.fromSameSingleAndPlural( 'approved' );
+
+// displays 'approved' in both cases
+console.log( approvedLabel.singular );
+console.log( approvedLabel.plural );
+```
+
+### _Properties_
+#### `Label.FORMAT_LOWERCASE`
+
+This static property on `Label` can be used to reference the lowercase format string.
+
+#### `Label.FORMAT_UPPERCASE`
+
+This static property on `Label` can be used to reference the uppercase format string.
+
+#### `Label.FORMAT_SENTENCE_CASE`
+
+This static property on `Label` can be used to reference the sentence case format string.
+
+### Instance
```

Comment:
Is this heading necessary?

The structure is kinda weird...
The following are the main headings (`<h3>`) used in this doc:

- Constructor
- Methods
- Properties
- Instance
- Methods

I think it could be reduced and reorganized to just be:

- Constructor
- Properties
- Methods

and the line following this comment:

> The following methods and properties exist after instantiating a `Label`.

could maybe be moved to the bottom of the "Constructor" section as an intro to the next two sections.

Reply:
No.  In javascript the `constructor` object is what is passed around.  So the `Constructor` heading refers to that and what is accessible on that object.  The `instance` heading refers to the methods and properties available on the instance after the object is constructed.  So they are _deliberately_ two sections.

Reply:
ahh kk